File: 53.Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==0){
            return 0;
        }
        int maxSum = nums[0];
        int curSum = 0;
        for(int i=0;i<nums.size();i++){
            curSum += nums[i];
            maxSum = curSum>maxSum?curSum:maxSum;
            curSum = curSum<0?0:curSum;
        }
        return maxSum;
    }
};

File: 53.Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = pre = 0
        for n in nums:
            pre = max(pre + n, 0)
            res = max(res, pre)
        if res == 0:
            return max(nums)
        return res
